genetic.py

```diff
@@ -2,35 +2,46 @@
 from network_components import *
 from graph import Graph
 from util import safe_softmax
+import pandas as pd
+from prettytable import PrettyTable
+import datetime
 
 
 class Sample:
     _id = 0
 
-    def __init__(self, innovation_history_instance):
+    def __init__(self, innovation_history_instance, is_final=1):
```

Comment:
is_final + finalize() comments:

1. I couldn't find a flow where is_final = 0. Probably intended for the random walk right? If not relevant anymore, please remove parameter and merge finalize with CTOR to make things simpler.
2. If we do need is_final, make it: is_final=True since it is a boolean and not a counter.

Reply:
2. changed to boolean. 
1. yes, it's used in random_walk and forgot to update this version there:
```suggestion
    scores = self.fitness_function([Sample(h, False) for h in histories])
```